server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1908,11 +1923,10 @@ public IndexMetadata build() {
             }
 
             // fill missing slots in inSyncAllocationIds with empty set if needed and make all entries immutable
-            @SuppressWarnings({ "unchecked", "rawtypes" })
-            Map.Entry<Integer, Set<String>> denseInSyncAllocationIds[] = new Map.Entry[numberOfShards];
+            List<Set<String>> denseInSyncAllocationIds = new ArrayList<>(numberOfShards);
```

Comment:
Can we keep it as an array instead and use `List.of` below to convert to list and make immutable?

Reply:
Hi @henningandersen , unfortunately I would need to keep the `@SuppressWarnings({ "unchecked", "rawtypes" })` if I were to use a `Set<String>[]`, so that is why I preferred an ArrayList. If the immutability is the aim here, I can use the unmodifiableList() later on when the denseInSyncAllocationIds is passed to the IndexMetadata constructor. Please tell me if this is not fine.

Reply:
It's fine, though I would have a (personal) preference for keeping the `@SupressWarning` in this case.

Reply:
I think the previous Map.Entry code looked better, but with the new approach, it would be a bit annoying because I cannot create a generic array and would also need to cast it to Set<String>[].